### PR TITLE
[Tests] Require exact match in assert_raises_init_error

### PR DESCRIPTION
**Pull request by random-zebra**
Backports 
- bitcoin#12718

This completes the todo left for 
- #2449.

### CODE REVIEW

File: test/functional/test_framework/test_node.py [hunk 1]
```diff
@@ -212,8 +212,12 @@ def wait_until_stopped(self, timeout=BITCOIND_PROC_WAIT_TIMEOUT):
 
     def assert_start_raises_init_error(self, extra_args=None, expected_msg=None, *args, **kwargs):
         """Attempt to start the node and expect it to raise an error.
+
+        extra_args: extra arguments to pass through to bitcoind
+        expected_msg: regex that stderr should match when bitcoind fails
```
Comment (furszy) [inline test/functional/test_framework/test_node.py:217]: -> pivxd

**Review by furszy — COMMENTED** [review]
Nice 🤘 , left a minor comment only

File: test/functional/test_framework/test_node.py (hunk 1 above)
Comment (random-zebra) [reply test/functional/test_framework/test_node.py]: Added e6119ed182c1b658fb949d9efaee6fb8b31932eb to update all comments in the test suite

**Comment by random-zebra** [comment]
Backported bitcoin#12853 too, and added a scripted commit to update the comments still mentioning `bitcoind` in all test files.

**Review by furszy — DISMISSED** [review]
cool, utACK e6119ed182c1b658fb949d9efaee6fb8b31932eb .

**Comment by random-zebra** [comment]
Rebased on master, to include `feature_asmap.py` just merged (#2480).

**Review by furszy — APPROVED** [review]
re-utACK bbabef1dc290fc1943f87817f562ebfa121938d8

File: test/functional/test_framework/test_node.py [hunk 2]
```diff
@@ -209,6 +217,44 @@ def is_node_stopped(self):
...
+        expected_msg: regex that stderr should match when pivxd fails
+
+        Will throw if pivxd starts without an error.
+        Will throw if an expected_msg is provided and it does not match pivxd's stdout."""
+        with tempfile.SpooledTemporaryFile(max_size=2**16) as log_stderr:
+            try:
+                self.start(extra_args, stderr=log_stderr, *args, **kwargs)
+                self.wait_for_rpc_connection()
+                self.stop_node()
+                self.wait_util_stopped()
```
Comment (Fuzzbawls) [inline test/functional/test_framework/test_node.py:233]: typo, should be `self.wait_until_stopped()`

**Review by Fuzzbawls — COMMENTED** [review]
minor typo in `assert_start_raises_init_error()`, otherwise ACK

File: test/functional/test_framework/test_node.py (hunk 2 above)
Comment (random-zebra) [reply test/functional/test_framework/test_node.py]: :+1: fixed

Comment (furszy) [reply test/functional/test_framework/test_node.py]: Will be good to have the python-lint script catching this stuff.

**Review by furszy — APPROVED** [review]
no other changes aside from the typo, utACK 84768b5

**Review by Fuzzbawls — APPROVED** [review]
ACK 84768b5eaff26901935056ec63041f99935f944a

**Pull request merged**